client/lib/reader-post-flux-adapter/index.js

```diff
@@ -42,6 +42,16 @@ const fluxPostAdapter = Component => {
 		state = this.getStateFromStores( this.props );
 
 		updateState = ( newState = this.getStateFromStores() ) => {
+			// check to see if this new state is the same as the old state
+			if ( newState.posts.length === this.state.posts.length ) {
```

Comment:
shouldComponentUpdate is for dealing with props that someone else is passing you. this is for dealing with internal state you're setting on yourself.